### PR TITLE
Cart mock user

### PR DESCRIPTION
**Pull request by Kevin-Solomon**
# Features Implemented
- [x] add to cart.
- [x] remove individual items from the cart.
- [x] add individual items to cart.
- [x] remove the whole item from cart.

# Features that need to be implemented 
- [ ] debouncing or show loading when the item is being added to cart. 
- [ ] move to wishlist feature.

*to access these feature you must be logged in*
```email : adarshbalika@gmail.com```
```password : adarshBalika123```

### CODE REVIEW

File: src/context/cart/cartContext.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        `/api/user/cart/${_id}`,
+        { action: { type: 'decrement' } },
+        {
+          headers: { authorization: token.token },
+        }
+      );
+      cartDispatch({ type: 'DECREASE_QUANTITY', payload: response.data.cart });
+    }
+  };
+  const increaseQuantity = async _id => {
```
Comment (Jay-Gadhiya) [inline src/context/cart/cartContext.js]: you can give some space after function end for batter code redability

**Comment by Jay-Gadhiya** [comment]
Code looks clean 
add to cart & remove cart is working fine

**Comment by Jay-Gadhiya** [comment]
Here is some bug, when i click on logo i willl automatically logout and page will reload. and buy now button is not redirecting to the product listing page.

**Review by mohammed-sas — APPROVED** [review]
Code looks fine
Functionalities works as desired

**Review by Jay-Gadhiya — APPROVED** [review]

File: src/context/cart/cartContext.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  const { token } = useAuth();
+  const [cartState, cartDispatch] = useReducer(cartReducer, initialCartState);
+  useEffect(() => {
+    async function getCartItems() {
+      console.log(token.token);
+      try {
+        const response = await axios.get('/api/user/cart', {
+          headers: { authorization: token.token },
+        });
+        console.log(response);
```
Comment (AtulPant2704) [inline src/context/cart/cartContext.js]: Remove this console.log

File: src/context/cart/cartContext.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      } else {
+        const response = await axios.post(
+          '/api/user/cart',
+          { product },
+          {
+            headers: { authorization: token.token },
+          }
+        );
+        cartDispatch({ type: 'ADD_TO_CART', payload: response.data.cart });
+        console.log(response);
```
Comment (AtulPant2704) [inline src/context/cart/cartContext.js]: Here also.

**Review by AtulPant2704 — APPROVED** [review]
Everything is working fine as mentioned but just want to point out one thing in LMS it is mentioned to display `Go to Cart` on button once it has been added to the cart, you can work on it later on. Rest all looks good.

**Pull request merged**